src/consensus/aft/raft.h

```diff
@@ -883,9 +872,34 @@ namespace aft
           }
         }
 
+        size_t backup_ack_timeout_count = 0;
         for (auto& node : nodes)
         {
           node.second.last_ack_timeout += elapsed;
+          if (node.second.last_ack_timeout >= election_timeout)
+          {
+            LOG_FAIL_FMT(
+              "For node {}, last_ack_timeout: {}",
+              node.first,
+              node.second.last_ack_timeout);
+            backup_ack_timeout_count++;
+          }
+        }
+
+        if (backup_ack_timeout_count >= get_quorum(nodes.size()))
```

Comment:
Is there anything stopping someone from deploying an even number of nodes? In theory, this should be n-q+1 not q, where n is number of nodes and q is the quorum size.

Reply:
Deploying an even number of nodes is supported by CCF, even though it's preferable to deploy an odd number of nodes. In `raft.h`, `nodes` only keeps track of other nodes in the service and doesn't include ourselves. This means that on the primary node, `nodes` contains all the backup nodes. When a majority of them (or more) times out, this means that it's likely that they're electing their own new primary and that we should step down.

Examples:
- 2-node network, q=2, q(`nodes`)=1, when 1 backup times out, we step down
- 3-node network, q=2, q(`nodes`)=2, when 2 backups time out, we step down
- 4-node network, q=3, q(`nodes`)=2, when 2 backups time out, we step down
- 5-node network, q=3, q(`nodes`)=3, when 3 backups time out, we step down

Reply:
As discussed, we're keeping the same logic for a 2-node network, even if the primary knows that the partitioned backup cannot elect a new primary on its own. This is mostly for consistency logic for all network configuration cases.